torch/csrc/inductor/aoti_eager/kernel_holder.cpp

```diff
@@ -0,0 +1,283 @@
+#if !defined(C10_MOBILE) && !defined(ANDROID)
+#include <torch/csrc/inductor/aoti_eager/kernel_holder.h>
+
+#include <ATen/ATen.h>
+
+#include <ATen/core/dispatch/Dispatcher.h>
+#include <torch/csrc/PyInterpreter.h>
+#include <torch/csrc/autograd/python_variable.h>
+#include <torch/csrc/inductor/aoti_runner/model_container_runner_cpu.h>
+#ifdef USE_CUDA
+#include <torch/csrc/inductor/aoti_runner/model_container_runner_cuda.h>
+#endif
+#include <torch/csrc/jit/frontend/function_schema_parser.h>
+
+namespace torch::inductor {
+
+namespace {
+enum class IValueType : uint8_t {
+  Tensor,
+  TensorList,
+  OptionalTensorList,
+  Scalar,
+  Invalid,
+};
+
+static bool HandleTensor(
+    const c10::IValue& ivalue,
+    const c10::Device& device,
+    std::vector<at::Tensor>& inputs) {
+  inputs.push_back(ivalue.toTensor());
+  return true;
+}
+
+static bool HandleTensorList(
+    const c10::IValue& ivalue,
+    const c10::Device& device,
+    std::vector<at::Tensor>& inputs) {
+  for (const auto& item : ivalue.toListRef()) {
+    if (!item.isNone()) {
+      inputs.push_back(item.toTensor());
+    }
+  }
+  return true;
+}
+
+static bool HandleOptionalTensorList(
+    const c10::IValue& ivalue,
+    const c10::Device& device,
+    std::vector<at::Tensor>& inputs) {
+  return HandleTensorList(ivalue, device, inputs);
+}
+
+static bool HandleScalar(
+    const c10::IValue& ivalue,
+    const c10::Device& device,
+    std::vector<at::Tensor>& inputs) {
+  inputs.push_back(at::scalar_tensor(
+      ivalue.toScalar(),
+      c10::TensorOptions().device(device).dtype(ivalue.toScalar().type())));
+  return true;
+}
+
+typedef bool (*HandlerFunc)(
+    const c10::IValue& ivalue,
+    const c10::Device& device,
+    std::vector<at::Tensor>& inputs);
+std::unordered_map<IValueType, HandlerFunc> handlers_ = {
+    {IValueType::Tensor, &HandleTensor},
+    {IValueType::TensorList, &HandleTensorList},
+    {IValueType::OptionalTensorList, &HandleOptionalTensorList},
+    {IValueType::Scalar, &HandleScalar}};
+
+bool HandleIValue(
+    const c10::IValue& ivalue,
+    const c10::Device& device,
+    std::vector<at::Tensor>& inputs) {
+  IValueType ivalue_type = IValueType::Invalid;
+  if (ivalue.isTensor()) {
+    ivalue_type = IValueType::Tensor;
+  } else if (ivalue.isTensorList()) {
+    ivalue_type = IValueType::TensorList;
+  } else if (ivalue.isOptionalTensorList()) {
+    ivalue_type = IValueType::OptionalTensorList;
+  } else if (ivalue.isScalar()) {
+    ivalue_type = IValueType::Scalar;
+  }
```

Comment:
Why not just call the handler functions directly here?  That would be less code and faster since it avoids an unorered_map lookup.